pkg/codegen/java/gen.go

```diff
@@ -1559,8 +1564,12 @@ func (mod *modContext) genFunctions(ctx *classFileContext, addClass addClassMeth
 					baseClass:             "com.pulumi.resources.InvokeArgs",
 					propertyTypeQualifier: inputsQualifier,
 					properties:            fun.Inputs.Properties,
+					args:                  true,
 				}
-				return args.genInputType(ctx)
+
+				// make sure the input properties are outputs
+				propertiesAsOutputs := true
```

Comment:
I kept like this because when reading the code outside of IDE, you don't see the intent but I guess it is better to keep these things as part of `plainType`

Reply:
this is the problem with flags :)